presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyExpressions.java

```diff
@@ -81,6 +83,6 @@ public SimplifyExpressions(Metadata metadata, SqlParser sqlParser)
 
     private Expression rewrite(Expression expression, Rule.Context context)
     {
-        return rewrite(expression, context, metadata, sqlParser);
+        return rewrite(expression, context.getSession(), context.getSymbolAllocator(), metadata, sqlParser);
```

Comment:
* can you please run it twice here, to make sure that subsequent execution does not rewrite anything. 
 * Also I think it would be valuable to extract some kind of `BaseExpressionRewriteTest` which you could then use for desugar rewrites. (it can be done later with desugar)